virion.md

```diff
@@ -1,161 +1,171 @@
+The Virion standard aims to solve this problem from a static linking approach
+by "shading" the classes, i.e. refactoring the classes in libraries
+to a unique namespace owned by the plugin to avoid duplication.
 
-**Do not use DEVirion for non-debugging reasons**. DEVirion does not shade the virion classes, so multiple plugins using DEVirion may still cause compatibility problems, making the virion framework meaningless.
+## Library specification
 
-### Compiling a virion with Poggit
-A Poggit virion project can be declared in the .poggit.yml of a repo similar to a Poggit plugin project, with two additional project attributes:
+Virions are distributed as a normal composer library.
+It MUST also declare an additional `x-virion` attribute in the composer.json:
 
-```yaml
-type: library
-model: virion
+```json
+{
+  "other": "normal fields...",
+  "x-virion": {
+    "spec": "3.0",
+    "antigen": "Name\\Space"
+  }
+}
 ```
 
-Example:
-
-```yaml
-projects:
-  libasynql:
-    path: path/to/libasynql
-    type: library
-    model: virion
-```
+`antigen` is a unique namespace owned by the virion.
+All items declared by the virion MUST reside in this namespace
+(or its subnamespaces, e.g. `Name\Space\Child` but not `Name\SpaceSuffix`).
+All items *not* declared by the virion MUST NOT be under this namespace,
+and SHOULD avoid using the antigen as a substring in their fully-qualified paths
+(the latter is only a foolproof requirement to reduce misbehavior of shading tools).
+To avoid ambiguation, the antigen MUST have at least two parts.
 
-### Compiling a virion without Poggit
-No official tools have been created to compile a virion release directly, but build scripts compiling a virion release can be created easily.
+The `spec` is the version of the virion specification (i.e. this document).
+Do not change this field unless
+the library requires features in newer versions of the virion specification.
 
-Basically, a virion release has the same format as the plugin release, except with four differences:
-- `plugin.yml` becomes the `virion.yml`
-- `resources` are not supported (may be added in the future).
-- A file called `virion.php` and a file called `virion_stub.php` are included into the phar. These files can be [downloaded from the Poggit Git repo](https://github.com/poggit/poggit/tree/beta/assets/php/).
-  - The `virion_stub.php` provides a command-line user interface for virion injection
-  - The `virion.php` contains the code that actually injects the virion.
-- The phar stub should load the `virion.php` directly. This is an example valid stub:
+A virion MAY declare `require` dependencies that also declare the `x-virion` field.
+Non-virion libraries MUST NOT be included
```

Comment:
For clarification, does this mean not included in the plugin's composer manifest or the virion's manifest?

Reply:
both, with the exception of platform dependencies like `php` which should be ignored. I will clarify this.